sale_coupon_criteria_multi_product/models/sale_coupon_criteria.py

```diff
@@ -0,0 +1,40 @@
+# Copyright 2021 Tecnativa - David Vidal
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SaleCouponCriteria(models.Model):
+    _name = "sale.coupon.criteria"
+    _description = "Coupon Multi Product Criteria"
+
+    program_id = fields.Many2one(comodel_name="sale.coupon.program",)
+    rule_min_quantity = fields.Integer(
+        string="Minimum Quantity",
+        compute="_compute_rule_min_quantity",
+        store=True,
+        readonly=False,
+        help="Minimum required product quantity to get the reward",
+    )
+    product_ids = fields.Many2many(comodel_name="product.product", string="Products",)
+    repeat_product = fields.Boolean(
+        string="Repeat", help="Can product quantities count multiple times or not",
+    )
+
+    @api.depends("product_ids", "repeat_product")
+    def _compute_rule_min_quantity(self):
+        """Set the minimum quantity automatically to prevent errors when the rule
+        isn't set to no repeat"""
+        for criteria in self.filtered(lambda x: not x.repeat_product):
+            criteria.rule_min_quantity = len(criteria.product_ids)
```

Comment:
Do you really need a computed field for the constraint?

Reply:
Well, it really acts as an onchange that helps the user setting the right quantity when the repeat products setting is off